irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -307,7 +308,11 @@ namespace iroha {
           if (not maybe_block) {
             return fmt::format("Failed to fetch block {}", height);
           }
-          notifier_.get_subscriber().on_next(*std::move(maybe_block));
+
+          std::shared_ptr<const shared_model::interface::Block> block_ptr =
+              std::move(maybe_block.get());
+          notifier_.get_subscriber().on_next(block_ptr);
+          getSubscription()->notify(EventTypes::kOnBlock, block_ptr);
```

Comment:
I would suggest to parametrize StorageImpl with `std::function<void(Block)>` so that the class and tests do not depend on subscription engine.